addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/core/LxControlTimedSwitch.java

```diff
@@ -0,0 +1,123 @@
+
+import java.io.IOException;
+
+import org.openhab.binding.loxone.internal.core.LxJsonApp3.LxJsonControl;
+
+/**
+ * A timed switch type of control on Loxone Miniserver.
+ * <p>
+ * According to Loxone API documentation, a switch control is:
+ * <ul>
+ * <li>a virtual input of switch type
+ * <li>a push button function block
+ * </ul>
+ *
+ * @author Stephan Brunner
+ *
+ */
+public class LxControlTimedSwitch extends LxControl {
+
+    /**
+     * A name by which Miniserver refers to timed switch controls
+     */
+    private static final String TYPE_NAME = "timedswitch";
+
+    /**
+     * deactivationDelay - countdown until the output is deactivated.
+     * 0 = the output is turned off
+     * -1 = the output is permanently on
+     * otherwise it will count down from deactivationDelayTotal
+     */
+    private static final String STATE_DEACTIVATION_DELAY = "deactivationDelay";
+
+    /**
+     * Command string used to set timed switch to ON
+     */
+    private static final String CMD_ON = "On";
+    /**
+     * Command string used to set timed switch to OFF
+     */
+    private static final String CMD_OFF = "Off";
+
+    /**
+     * Create timed switch control object.
+     *
+     * @param client
+     *            communication client used to send commands to the Miniserver
+     * @param uuid
+     *            switch's UUID
+     * @param json
+     *            JSON describing the control as received from the Miniserver
+     * @param room
+     *            room to which switch belongs
+     * @param category
+     *            category to which switch belongs
+     */
+    LxControlTimedSwitch(LxWsClient client, LxUuid uuid, LxJsonControl json, LxContainer room, LxCategory category) {
+        super(client, uuid, json, room, category);
+    }
+
+    /**
+     * Check if control accepts provided type name from the Miniserver
+     *
+     * @param type
+     *            name of the type received from Miniserver
+     * @return
+     *         true if this control is suitable for this type
+     */
+    public static boolean accepts(String type) {
+        return type.equalsIgnoreCase(TYPE_NAME);
+    }
+
+    /**
+     * Set timed switch to ON.
+     * <p>
+     * Sends a command to operate the timed switch.
+     *
+     * @throws IOException
+     *             when something went wrong with communication
+     */
+    public void on() throws IOException {
+        socketClient.sendAction(uuid, CMD_ON);
+    }
+
+    /**
+     * Set timed switch to OFF.
+     * <p>
+     * Sends a command to operate the timed switch.
+     *
+     * @throws IOException
+     *             when something went wrong with communication
+     */
+    public void off() throws IOException {
+        socketClient.sendAction(uuid, CMD_OFF);
+    }
+
+    /**
+     * Get current value of the timed switch'es state.
+     *
+     * @return
+     *         0 - switch off, 1 - switch on
+     */
+    public Double getState() {
```

Comment:
Additionally to this method there could be a getDeactivationDelay() (stupid name following the API, maybe some other name) which would return the countdown to the switch off. You could add another Number channel to this control to pass this value to the openHAB. Then I could imagine building a button in habPanel, which after pressing shows the realtime countdown until switch is off. How cool would that be.

Reply:
@ppieczul I have created a second channel for timed switch see

 if (control instanceof LxControlTimedSwitch) {
                ChannelUID id2 = getChannelIdForControl(control, 1);
                addChannel(channels, "Number", switchTypeId, id2, label + " / Decativiation Delay",
                        "Deactiviation Delay", tags);
            }

I don't really like how this is made, but I don't wanted to change the structure of the code, only because I wanted to create a second channel for on LxControl. I'd prefer to return the list of the channel as an abstract method of LxControl. What do you think?

Reply:
The .core package is really a code that I would like to keep completely separated from openhab, so it could be an independent java library for Loxone. Today this is not exported, but it may become either exported or separated in the future.

Reply:
got it. fair point

Reply:
See remark about using int's for the return in the earlier comment